libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/arrow/api/MapboxRouteArrowView.kt

```diff
@@ -47,7 +55,10 @@ class MapboxRouteArrowView(private val options: RouteArrowOptions) {
      * @param visibilityChange a state containing data for applying the view side effects.
      */
     fun render(style: Style, visibilityChange: ArrowVisibilityChangeValue) {
-        initializeLayers(style, options)
+        if (rebuildLayersOnFirstRender || !layersAreInitialized(style)) {
+            rebuildLayersOnFirstRender = false
+            rebuildLayersAndSources(style, options)
+        }
```

Comment:
This block of code can be moved to a function, like: `fun rebuildSourcesAndLayersIfNeeded()`.

Reply:
done